### PR TITLE
[0.60] DropViews still in ShadowNode registry during UIManager dtor

### PR DESCRIPTION
**Pull request by FalseLobster**
Office E2E Automation is running into a failure during shutdown caused by rapidly spinning up & shutting down some react-native UI.  What's happening is some Views get created, but the batch to add them to the ui tree never completes.  Usually the removeAllRootView call is sufficient to tear down the UI, but since the shadow nodes finished their batch, they're not parented to the root view.  This iterates through all shadow node left in the registry after removing the root view and explictly calls DropView on them so they can clean up their resources.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5854)

### CODE REVIEW

File: vnext/ReactWindowsCore/Modules/UIManagerModule.cpp [hunk 1]
```diff
@@ -26,7 +26,9 @@ UIManager::UIManager(std::vector<std::unique_ptr<IViewManager>> &&viewManagers,
 
 UIManager::~UIManager() {
   m_nodeRegistry.removeAllRootViews([this](int64_t rootViewTag) { removeRootView(rootViewTag); });
-
+  for (auto it = m_nodeRegistry.getAllNodes().begin(); it != m_nodeRegistry.getAllNodes().end(); it++) {
```
Comment (asklar) [inline vnext/ReactWindowsCore/Modules/UIManagerModule.cpp:29]: I'd rather have us give the m_nodeRegistry a lambda to execute over its elements than to tightly couple it to the type of a private member, so maybe something like:
```cpp
m_nodeRegistry.ForAllNodes([] (int tag, shadow_ptr const&) {  DropView(tag, false, true); });
```

**Comment by NickGerleman** [comment]
Could you please also make this change in master/0.62? We'll otherwise see the same issue pop up again 😕.

**Comment by FalseLobster** [comment]
> Could you please also make this change in master/0.63/0.62? We'll otherwise see the same issue pop up again 😕.

Yup!  That's the plan

**Comment by Khalef1** [comment]
@FalseLobster : Thanks for working on this! Note that the change also needs to be backported to RNW 0.60 and into previous Office forks (I believe July and June, but please confirm with @tudorms or @NikoAri ) .

**Comment by FalseLobster** [comment]
> @FalseLobster : Thanks for working on this! Note that the change also needs to be backported to RNW 0.60 and into previous Office forks (I believe July and June, but please confirm with @tudorms or @NikoAri ) .

Why does it need to be backported to our June/July forks?

**Comment by Khalef1** [comment]
Those are the forks we are scaling out to prod currently and where we are blocked

File: vnext/ReactWindowsCore/ShadowNodeRegistry.h [hunk 2]
```diff
@@ -32,6 +33,8 @@ struct ShadowNodeRegistry {
 
   ShadowNode *getParentRootShadowNode(int64_t nodeTag);
 
+  void ForAllNodes(const Mso::FunctorRef<void(int64_t, shadow_ptr const &)> &fnDo) noexcept;
```
Comment (asklar) [inline vnext/ReactWindowsCore/ShadowNodeRegistry.h:36]: Doesn't the functor need to be marked noexcept too?

**Pull request merged**